algoliasearch/src/main/java/com/algolia/search/saas/BaseAPIClient.java

```diff
@@ -96,6 +98,26 @@ protected BaseAPIClient(String applicationID, String apiKey, List<String> hostsA
         headers = new HashMap<String, String>();
     }
 
+    public void enableSearchCache() {
+        isCacheEnabled = true;
+        if (cache == null) {
+            cache = new ExpiringCache<>();
+        }
+    }
+
+    public void enableSearchCache(int timeoutInSeconds, int maxRequests) {
+        isCacheEnabled = true;
+        cache = new ExpiringCache<>(timeoutInSeconds, maxRequests);
+    }
+
+    public void disableSearchCache() {
+        isCacheEnabled = false;
+        if (cache != null) {
+            cache.reset();
+            cache = null;
+        }
+    }
```

Comment:
I would also add a `clearCache`, for convenience purpose (instead of disable+enable).